### PR TITLE
Name Request Correction Form Validators  Update

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5082

*Description of changes:*

* Simplified and  updated the form validations for phone and email on name request search. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/YourCompany/CompanyName/CorrectNameRequest.vue [hunk 1]
```diff
@@ -94,12 +94,12 @@ export default class CorrectNameRequest extends Mixins(NameRequestMixin) {
     v => this.validateNameRequestNumber(v) || 'Name Request Number is invalid'
   ]
   private entityPhoneNumberRules = [
-    v => this.isInputEntered(v, 'phone') || 'Phone number is required',
+    v => (v !== '' || !this.entityEmail) || 'Phone number is required',
```
Comment (severinbeauvais) [inline src/components/YourCompany/CompanyName/CorrectNameRequest.vue:97]: This looks backwards... please add a comment that it's one or the other, or something :)

**Review by severinbeauvais — APPROVED** [review]
Okey doke.

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/components/YourCompany/CompanyName/CorrectNameRequest.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue:97]: oh no your right, i can do this better. Will refactor!

**Review by lekshmimallika-aot — APPROVED** [review]

File: src/components/YourCompany/CompanyName/CorrectNameRequest.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue]: Ok, fixed my wonky rules. They didn't make any sense. Refactored and Simplified :P

Comment (severinbeauvais) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue]: Does the new code still enforce "either phone number or email"?

Comment (severinbeauvais) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue]: And do you have a unit test for this, so we find out right away if we break it in the future?

Comment (cameron-freshworks) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue]: Yes, the form itself will never be valid unless one or the other are present.
No, will add one!

Comment (cameron-freshworks) [reply src/components/YourCompany/CompanyName/CorrectNameRequest.vue]: I can explain as well, this fix was more of a UX fix as the core of the validation wasn't broken prior. 

There is two layers of validation, the form itself and the input fields individually. Prior, the input fields were validating as if they were required, which technically neither is required, but one or the other needs to be there to complete ( validating at the form level ). So i removed the 'required rule' at the input level, and just relied on the form validation to check at least one of the values is present.

**Pull request merged**